ops/model.py

```diff
@@ -1595,27 +1729,38 @@ class _ModelBackend:
     """
 
     LEASE_RENEWAL_PERIOD = datetime.timedelta(seconds=30)
-
-    def __init__(self, unit_name=None, model_name=None, model_uuid=None):
-        if unit_name is None:
-            self.unit_name = os.environ['JUJU_UNIT_NAME']
-        else:
-            self.unit_name = unit_name
-        if model_name is None:
-            model_name = os.environ.get('JUJU_MODEL_NAME')
-        if model_uuid is None:
-            model_uuid = os.environ.get('JUJU_MODEL_UUID')
-        self.model_name = model_name
-        self.model_uuid = model_uuid
-        self.app_name = self.unit_name.split('/')[0]
-
-        self._is_leader = None
+    _STORAGE_KEY_RE = re.compile(
+        r'.*^-s\s+\(=\s+(?P<storage_key>.*?)\)\s*?$',
+        re.MULTILINE | re.DOTALL
+    )
+
+    def __init__(self, unit_name: Optional[str] = None,
+                 model_name: Optional[str] = None,
+                 model_uuid: Optional[str] = None):
+        def _load_from_env_if_none(value: Optional[str], env_key: str,
+                                   check_not_none: bool = False) -> str:
+            if value is None:
+                value_ = os.environ.get(env_key)
+                if check_not_none:
+                    assert value_ is not None
+                return typing.cast(str, value_)
+            return value
+        self.unit_name = _load_from_env_if_none(unit_name, 'JUJU_UNIT_NAME', check_not_none=True)
+        self.model_name = _load_from_env_if_none(model_name, 'JUJU_MODEL_NAME')
+        self.model_uuid = _load_from_env_if_none(model_uuid, 'JUJU_MODEL_UUID')
+        self.app_name = self.unit_name.split('/')[0]  # type: str
```

Comment:
```python
self.unit_name = unit_name or os.getenv('JUJU_UNIT_NAME')
self.model_name = model_name or os.getent('JUJU_MODEL_NAME')
...
```

seems maybe a bit nicer. And probably a raise would be preferable for the unit name check over an assert.

Reply:
The problem with that is that the type checker will complain if I say that `self.unit_name` is a `str`, since os.getenv returns `Optional[str]`. Will think of a cleaner solution.